docs/ref/templates/builtins.txt

```diff
@@ -1628,6 +1628,38 @@ For example::
 If ``value`` is ``"testing\r\njavascript \'string" <b>escaping</b>"``,
 the output will be ``"testing\\u000D\\u000Ajavascript \\u0027string\\u0022 \\u003Cb\\u003Eescaping\\u003C/b\\u003E"``.
 
+.. templatefilter:: json_tag
+
+``json_tag``
+------------
+
+.. versionadded:: 2.1
+
+Outputs an object JSON-encoded, wrapped in a <script> tag ready for use with JavaScript.
+
+**Argument:** HTML "id" of the <script> tag.
+
+For example::
+
+    {{ value|json_tag:"hello-data" }}
+
+If ``value`` is a the dictionary ``{'hello': 'world'}``, the output will be::
+
+    <script id="hello-data" type="application/json">{"hello": "world"}</script>
+
+The resulting data can be accessed in JavaScript like this::
+
+    var el = document.getElementById('hello-data');
+    var value = JSON.parse(el.textContent || el.innerText);
+
+Possible XSS attacks are mitigated by escaping the characters "<", ">" and "&". For example if ``value`` is ``{'hello': 'world</script>&amp;'}``, the filter's output is::
+
+    <script id="hello-data" type="application/json">{"hello": "world\\u003C/script\\u003E\\u0026amp;"}</script>
+
+This is compatible with a strict Content Security Policy that prohibits all
+in-page script execution and maintains a clean separation between passive data
+and executable code.  See https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#RULE_.233.1_-_HTML_escape_JSON_values_in_an_HTML_context_and_read_the_data_with_JSON.parse for more details.
```

Comment:
Please don't put raw URLs in text. That being said, there's a comment from Florian in the function's docstring that questions if this URL should be referenced at all.